lib/galaxy/util/__init__.py

```diff
@@ -173,6 +173,24 @@ def directory_hash_id(id):
         return list(iter(s[0:3]))
 
 
+def find_dataset(path):
+    """
+    Perform DFS starting at path. Return the first occurrence of a dataset.
+    `path` should be a directory.
+    Return a DirEntry object or None.
+    """
+    if not os.path.exists(path) or not os.path.isdir(path):
+        return
+    stack = [path]
+    while stack:
+        with os.scandir(stack.pop()) as iter:
```

Comment:
`iter` is a standard library function, can you please use a different variable name?